skiko/src/commonMain/kotlin/org/jetbrains/skia/Typeface.kt

```diff
@@ -88,7 +89,16 @@ class Typeface internal constructor(ptr: NativePointer) : RefCnt(ptr) {
     val variations: Array<FontVariation>?
         get() = try {
             Stats.onNativeCall()
-            _nGetVariations(_ptr)
+            val count = _nGetVariationsCount(_ptr)
+            if (count > 0) {
+                val variationsData = withResult(IntArray(count * 2)) {
+                    _nGetVariations(_ptr, it, count)
+                }
+                (0 until count).map { i ->
```

Comment:
Maybe https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/for-each-indexed.html on variationsData would be make code a bit more readable?